addons/website_payment/controllers/portal.py

```diff
@@ -84,6 +84,16 @@ def donation_transaction(self, amount, currency_id, partner_id, access_token, mi
 
         return tx_sudo._get_processing_values()
 
+    @staticmethod
+    def _update_landing_route(tx_sudo, access_token):
+        # the user can change the donation amount on the payment page,
+        # therefor we need to recompute the access_token
+        if tx_sudo.is_donation:
+            access_token = payment_utils.generate_access_token(
+                tx_sudo.partner_id.id, tx_sudo.amount, tx_sudo.currency_id.id
+            )
+        return super(PaymentPortal, PaymentPortal)._update_landing_route(tx_sudo, access_token)
```

Comment:
`(PaymentPortal, PaymentPortal)` ?

Reply:
Bounded super call on a static method.
(For more info: https://bugs.python.org/issue31118)

The parent method is static, this is how we need to call it in python.
`super()` implicitly resolves to `super(class, instance)`, and since the function is static, not a method, the lookup for the function fails.

Yeah I know it is not the prettiest thing :S but there is no other alternative.

Reply:
Does it really makes sense to have flagged that method as `staticmethod` in the first place? Is there any reason/gain to really to it?

Globally speaking, shouldn't a method either be a pure util method and then defined as module level function, or otherwise be defined as a method of a class but without `staticmethod` decorator?
And for model classes, at best flagged with api.model decorator (even if useful only if callable from rpc which is not the case here as private (and through xml data but unlikely here))

As this is, if I am not mistaken, the first occurrence in our whole codebase of such a `super(Class, Class)`, I am wondering.
Even tho this is stable so you don't have a choice here I guess.

Reply:
Now for the reason, I can only guess. It was introduced by https://github.com/odoo/odoo/commit/7fccbac004628093da49016f75370a84bba49465 by @bso-odoo almost 2 years ago, if he remembers? I assume it's so it can be overriden, and since self is not necessary, we flag it as static. I see that the controller has quite a few staticmethods.
Actually there are other cases of bounded super call for static methods, but those are under the guise of `super(class, cls)` in functions tagged @classmethod. But yeah I think I guess this patch is the first introduction of a `super(Class, Class)` (unless my regex was not correct).

Reply:
IMHO `@staticmethod` is inappropriate if you expect it to be overridden.
And even if `self` is not used, it could be a simple method.
A class method is only useful if you intend to make it invokable on the class itself.
And by the way, in any case, you can use `super()` without parameters, it is more readable.

Reply:
If the method is tagged with `@staticmethod`, calling `super()` without parameters results in a `TypeError: super(type, obj): obj must be an instance or subtype of type`. (since there is no self in the method, then obj = first argument of the function, which is `tx_sudo`, which indeed is not an instance of `PaymentPortal`) (more info: help(super) in a python REPL).

I could remove the staticmethod tag from the method I introduced and add a self as first param, then `super()` without parameters works. The con is that then the function has a different signature than the parent. 

IMHO I would prefer to keep the same signature at the cost of the ugly `super(Class, Class)` call, and afterward if there is a future task to refactor this controller to remove all the `@staticmethods`, then update the `super()` call.